File: 프로그래머스/lv1/92334. 신고 결과 받기/신고 결과 받기.py
def solution(id_list, report, k):
    ID_dict = {}
    L = len(id_list)
    call = [[0] * L for _ in range(L)]
    
    for i in range(L):
        ID_dict.setdefault(id_list[i],i)
        
    for FT in report:
        F,T = FT.split()
        call[ID_dict[T]][ID_dict[F]] = 1 
    
    result = [0] * L
    
    for num,x in enumerate(id_list,start=1):
        print(num,x)
    
    
    for lst in call:
        if lst.count(1) >= k:
            for i in range(L):
                if lst[i] == 1:
                    result[i] +=1    
    return result

File: 프로그래머스/1/92334. 신고 결과 받기/신고 결과 받기.js
function solution(id_list, report, k) {
    let N = id_list.length
    const id_dict = {}
    id_list.forEach((value,idx) => id_dict[value] = idx )
    G = Array.from({length : N}, () => Array(N).fill(0))
    
    report.forEach((name,idx) => {
        const [report, receive] = name.split(" ")
        G[id_dict[report]][id_dict[receive]] = 1
    })
    
    const result = []
    for(let i = 0; i < N ; i ++ ){
        let people = 0
        for(let j = 0; j < N ; j ++ ){
            if (G[i][j] === 1) {
                let cnt = 0
                for (let l = 0; l < N; l++){
                    cnt += G[l][j]
                }
                if (cnt >= k) {people ++}
            }
        }
        result.push(people)
    }
    return result;
}